lib/utils.js

```diff
@@ -171,15 +176,17 @@ var request = function request(options, data, timeout) {
         res.payload = chunks.join('');
 
         // Validate content-length if one was provided
+        // TODO - getBlobProperties returns content-length (the content length of the blob),but it does not have a payload!!
+
         var contentLength = res.headers['content-length'];
-        if (contentLength) {
-          var length = Buffer.byteLength(res.payload, 'utf8');
-          if (length !== parseInt(contentLength)) {
-            var err = new Error('Content-Length mismatch');
-            err.code = 'RequestContentLengthError';
-            return reject(err);
-          }
-        }
+        // if (contentLength) {
```

Comment:
Let's make it a `if (contentLength && options.method !== 'HEAD') {`

that should solve the TODO above... as getBLobProperties is a HEAD request:
https://docs.microsoft.com/en-us/rest/api/storageservices/fileservices/get-blob-properties

Per HTTP spec `HEAD` requsts can't contain a body. So checking content-lenght of it wouldn't make any sense.